Framework/include/QualityControl/CustomParameters.h

```diff
@@ -0,0 +1,213 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file   CustomParameters.h
+/// \author Barthelemy von Haller
+///
+
+#ifndef QC_CUSTOM_PARAMETERS_H
+#define QC_CUSTOM_PARAMETERS_H
+
+#include <string>
+#include <unordered_map>
+#include <stdexcept>
+#include <optional>
+
+namespace o2::quality_control::core
+{
+
+/**
+ * This class represents the parameters provided by the users in their config file.
+ * A value can be defined for a specific run type and/or beam type. It can also be set for any
+ * run type or beam type, in such a case we use the special runType and beamType `default`.
+ * We expect the strings runType and beamType to correspond to what is provided by the Bookkeeping.
+ *
+ * Example:
+ *   CustomParameters cp;
+ *   cp.set("key", "value"); // for the default run and beam type
+ *   cout << "value for key `key` : " << cp.at("key") << endl;
+ *
+ *   CustomParameters cp2;
+ *   cp2.set("key", "value_run1_beam1", "physics", "pp");
+ *   cout << "value for key `key` : " << cp.at("key", "physics", "pp") << endl;
+ *
+ */
+class CustomParameters
```

Comment:
I don't quite understand why we should recompile these methods for each time someone includes this header. Perhaps everything could be moved to a source file?

Reply:
ok I had missed your previous comment
fixed.